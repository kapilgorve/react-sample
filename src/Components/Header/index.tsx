import React, { Component } from 'react';
import styled from 'styled-components';

const HeaderContainer = styled.div`
background-color: #1cbbb4;
padding: 4rem;
text-align: center;
`;

const HeaderTitle = styled.h2`
font-size: 40px;
color: white;
font-weight: lighter;
margin: 0;
`;

const Sub = styled.h4`
font-size: 20px;
color: #ffffffc7;
font-weight: lighter;
margin: 0;
`;


class Header extends Component {
  render() {
    return (
      <HeaderContainer>
        <HeaderTitle>EMI Calculator</HeaderTitle>
        <Sub>Calculate your Loan EMI & Total Interest Due</Sub>
      </HeaderContainer>
    );
  }
}

export default Header;
