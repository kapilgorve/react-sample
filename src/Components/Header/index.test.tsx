import React from 'react';
import { render , cleanup } from '@testing-library/react';
import Header from './index';

afterEach(cleanup)


test('renders header', () => {
  const { getByText } = render(<Header />);
  const linkElement = getByText(/EMI Calculator/i);
  expect(linkElement).toBeInTheDocument();
});

describe('<Header/> spec', () => {
  it('renders header component', () => {
      const {asFragment} =  render(<Header/>)
      expect(asFragment()).toMatchSnapshot();
  });
});
