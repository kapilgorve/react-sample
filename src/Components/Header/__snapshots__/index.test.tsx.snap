// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> spec renders header component 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm CNnaw"
  >
    <h2
      class="sc-AxirZ Cttsw"
    >
      EMI Calculator
    </h2>
    <h4
      class="sc-AxiKw HZTEe"
    >
      Calculate your Loan EMI & Total Interest Due
    </h4>
  </div>
</DocumentFragment>
`;
